src/main/scala/org/broadinstitute/dsde/vault/services/uBAM/IngestServiceHandler.scala

```diff
@@ -81,10 +93,17 @@ case class IngestServiceHandler(requestContext: RequestContext, bossService: Act
   def completeIfDone =
     dmId match {
       case Some(id) =>
-        if (fileCount == bossURLs.size) {
-          log.debug("uBAM ingest complete")
-          requestContext.complete(UBamIngestResponse(id, bossURLs).toJson.prettyPrint)
-          context.stop(self)
+        if (fileCount == bossObjects.size) {
```

Comment:
This is the done condition for the `X-Force-Location` path but the proper API needs to check `bossURLs.size` instead.  Move those checks inside the `case` statements.